test/app1/package.json

```diff
@@ -17,7 +17,7 @@
         "package_name": "{node_abi}-{platform}-{arch}.tar.gz"
     },
     "scripts": {
-        "install": "node-pre-gyp install --fallback-to-build",
+        "install": "${node_pre_gyp_command:-npx node-pre-gyp} install --fallback-to-build",
```

Comment:
[question] what this is change doing? It is optional or needed?

Reply:
that's no longer there though i replaced it with a `test/tape-helper.js` which works fine on linux. just pushed; we'll see how it works on windows. i'll need to rethink a few other pieces as well but wanted to give it a shot.

maybe i'm trying to hard not to put test-specific changes in `install.js`. in order to mock http without modifying `install.js` the only solution i came up with was to change the command `node-pre-gyp install`  to `node -r create-mock-s3 node-pre-gyp install`. that loads the create-mock-s3 module (in test, so not even distributed via npm) so doesn't require changes to `install.js` or additional code to be distributed. so detecting that environment variable is required. but my most recent commit moves that logic to `test/tape-helper.js` so no linux-specific syntax is in the script section.

Reply:
i do not know how to get around this in the test apps without requiring changes to `install.js` for mocking http. i'm guessing you're using https because it's more universal than s3 and it doesn't require aws-sdk to be installed, but one solution is just to move to using s3. it'll be a bigger distribution though.

another solution is to just change the test commands to execute what `npm install` would execute directly. it bypasses package.json but tests the same thing, i believe.

the other solution i can think of is to modify `install.js` to load the mocking module if the `node_pre_gyp_mock_s3` environment variable is present. i avoided that in favor of not modifying "production" code, but it's doable and not a big deal.

Reply:
@springmeyer - here's a better idea of the options i see. the first three do not require any changes to `install.js` for https mocking.

1 use what it's doing now but fix it for windows. that requires yet another `tape-helper`-like module to spawn the tests after modifying the command line. the primary advantage is that there is no additional code in `install.js`; the primary negative is that it mucks around with the actual commands, i.e., we're not testing the exact same thing the user is executing.
2 move to using s3 to install. advantage - no need to mock https in install. negative - requires aws-sdk for install, compatibility not likely to be as good as https.
3 just modify the test commands directly, i.e. instead of `build.test.js` executing `npm install --fallback-to-build=false` it just executes what `npm install` actually does, `node-pre-gyp install`. advantage - no changes to package.json, disadvantage - doesn't use package.json.
4 modify `install.js`. i had been trying to avoid this but i think that goal was a bit off. i'd been thinking of this as something that runs in "production" as opposed to just during installation. there is very little reason not to change `install.js` to add mocking when in testing mode.

if you're ok with 4, i'll do that. that means some of what i did was wasted but i think it's the better solution.